script/generate-documentation

```diff
@@ -35,20 +46,23 @@ update_docs() {
         # gh#1480
         # 2 files changed, 2 insertions(+), 2 deletions(-)
         ANYTHING_CHANGED=$(git diff --shortstat | perl -ne 'my $n; print $n = () = m/(?: 2 \w+)/g')
-        if [ "${ANYTHING_CHANGED}" -ne 3 ]; then
+        if [ "${ANYTHING_CHANGED}" -ne 3 ] && [ -n "${ANYTHING_CHANGED}" ]; then
             cd ..
             git add _includes/api.html
             git add docs/index.html docs/current.pdf docs/api/testapi.html
             git add docs/api/schema
-            echo "Update documentation to commit ${shortref}" > last.commit
+            topic="Update documentation to commit ${shortref}"
+            echo "$topic" > last.commit
             echo "" >> last.commit # somehow travis does not like \n
             [[ -z "${CIRCLE_SHA1}" ]] || (cd .. && git log --pretty=fuller "${CIRCLE_SHA1}" -1 >> out/last.commit)
             git commit -F last.commit
-            [[ -n ${publish} ]] && git push "$SSH_REPO" "$TARGET_BRANCH"
+            [[ -z "${PUBLISH}" ]] || {
```

Comment:
That's a bit confusing to read imho. `if [[ -n ${PUBLISH} ]]....` would avoid the double negative